querysync/java/com/google/idea/blaze/qsync/project/ProjectDefinition.java

```diff
@@ -85,6 +90,11 @@ public QuerySpec deriveQuerySpec(Context<?> context, Path workspaceRoot) throws
       }
     }
     for (Path exclude : projectExcludes()) {
+     if(systemExcludes().contains(exclude.toString())){
+       // We don't have to check if these dirs are valid for queries, because they also don't fall into //... wildcard
+       continue;
```

Comment:
but do we still need to exclude them via `result.excludePath(exclude);` or not?

Reply:
I'm not sure, can investigate it later. The whole point of this PR is to avoid `isValidPathForQuery` for these paths which might be super expensive

Reply:
So that's basically an optimization, it doesn't change the behavior

Reply:
what happens if we added `result.excludePath(exclude);` instead of `continue`? if we get the same performance improvement then it seems like the right thing to do

Reply:
I haven't tried but I think it could go to the query, as a substitution unconditionally, without validity check and hence could result in a crash

Reply:
I see that makes sense, I can merge it as is to release for 2024.3 but we may come up with a better optimization later